src/os/ui/ui.js

```diff
@@ -203,7 +203,7 @@ os.ui.apply = function(scope, opt_delay) {
   } else {
     try {
       if (scope && (!scope.$root || !scope.$root.$$phase)) {
-        scope.$apply();
+        scope.$applyAsync();
```

Comment:
I like this alot more. I'm not sure you need the if statement anymore right? the purpose of that was to check if its in a digest to not call apply

Reply:
I do like the convenience of a built-in null check, similar to `goog.dispose`. That was initially to protect against attempts to call apply after the controller was destroyed and scope nulled without wrapping in an explicit `if (this.scope)`.